src/presenter/presenter.js

```diff
@@ -2,40 +2,79 @@ import TripSortView from '../view/trip-sort-view.js';
 import TripEventsListView from '../view/trip-events-list-view.js';
 import EventFormView from '../view/event-form-view.js';
 import EventView from '../view/event-view.js';
-import { render } from '../render.js';
+import { render, replace } from '../framework/render.js';
+import NoEventView from '../view/no-event-view.js';
 
 export default class BoardPresenter {
-  sortComponent = new TripSortView();
-  eventListComponent = new TripEventsListView();
+  #boardContainer = null;
+  #eventsModel = null;
+  #sortComponent = new TripSortView();
+  #eventListComponent = new TripEventsListView();
+
+  #boardEvents = [];
 
   constructor({boardContainer, eventsModel}) {
-    this.boardContainer = boardContainer;
-    this.eventsModel = eventsModel;
+    this.#boardContainer = boardContainer;
+    this.#eventsModel = eventsModel;
   }
 
   init() {
-    this.boardEvents = [...this.eventsModel.getTasks()];
-    this.destinations = this.eventsModel.getDestinations();
-    this.offers = [...this.eventsModel.getOffers()];
-
-    render(this.sortComponent, this.boardContainer);
-    render(this.eventListComponent, this.boardContainer);
-    render(new EventFormView({
-      event: this.boardEvents[0],
-      // destinations: this.eventsModel.getDestinationById(this.boardEvents[0].destination),
-      destinations: this.destinations,
-      // offers: this.eventsModel.getOffersByType(this.boardEvents[0].type)
-      offersByType: this.eventsModel.getOffersByType(this.boardEvents[0].type)
-    }),
-    this.eventListComponent.getElement());
-
-    for (let i = 0; i < this.boardEvents.length; i++) {
-      render(new EventView({
-        event: this.boardEvents[i],
-        destination: this.eventsModel.getDestinationById(this.boardEvents[i].destination),
-        offersByType: this.eventsModel.getOffersByType(this.boardEvents[i].type)
-      }),
-      this.eventListComponent.getElement());
+
+    this.#boardEvents = [...this.#eventsModel.events];
+    this.destinations = [...this.#eventsModel.destinations];
+    this.offers = [...this.#eventsModel.offers];
+
+    this.#renderBoard();
+  }
+
+  #renderEvent(event, allDestinations, offersByType) {
+    allDestinations = this.destinations;
+    offersByType = this.#eventsModel.getOffersByType(this.#boardEvents[0].type);
+    const escKeyDownHandler = (evt) => {
+      if (evt.key === 'Escape') {
+        evt.preventDefault();
+        replaceFormToCard();
+        document.removeEventListener('keydown', escKeyDownHandler);
+      }
+    };
+    const eventComponent = new EventView({
```

Comment:
Отделяем блоки кода 1 строкой
```suggestion
    };
    
    const eventComponent = new EventView({
```